File: cn/java-spring-webflux-demo/src/main/java/com/itangcent/springboot/webflux/demo/controller/UserCtrl.java
package com.itangcent.springboot.webflux.demo.controller;

import com.itangcent.common.dto.IResult;
import com.itangcent.common.dto.Result;
import com.itangcent.common.model.UserInfo;
import org.springframework.web.bind.annotation.*;
import reactor.core.publisher.Mono;

import java.util.Collections;

/**
 * 用户管理控制器
 * 提供用户信息的增删改查等基本操作
 *
 * @module user
 */
@RestController
@RequestMapping(value = "user")
public class UserCtrl {

    /**
     * 欢迎接口
     * 返回简单的问候信息
     *
     * @return 问候信息
     */
    @RequestMapping(value = "/index")
    public Mono<String> greeting() {
        return Mono.just("hello world");
    }

    /**
     * 更新用户名称
     * 根据用户ID更新用户的名称和其他信息
     *
     * @param id 用户ID
     * @param newName 新的用户名
     * @param slogon 个性签名
     * @param times 操作次数
     * @return 更新后的用户信息
     * @deprecated 请使用 {@link #update(UserInfo)} 替代
     */
    @RequestMapping(value = "/set", method = RequestMethod.PUT)
    public Mono<Object> set(long id,
                            @RequestParam String newName,
                            @RequestParam(required = false, defaultValue = "haha") String slogon,
                            @RequestParam(required = false, defaultValue = "10") long times) {

        UserInfo userInfo = new UserInfo();
        userInfo.setId(id);
        userInfo.setName(newName);
        userInfo.setAge(45);
        return Mono.just(Result.success(userInfo));
    }

    /**
     * 获取用户信息
     * 根据用户ID查询用户的详细信息
     *
     * @param id 用户ID
     * @return 用户信息
     * @deprecated 此接口已废弃，请使用新版本的查询接口
     */
    @Deprecated
    @GetMapping("/get/{id}")
    public Mono<IResult> get(@PathVariable("id") Long id) {
        UserInfo userInfo = new UserInfo();
        userInfo.setId(id);
        userInfo.setName("Tony Stark");
        userInfo.setAge(45);
        return Mono.just(Result.success(userInfo));
    }

    /**
     * 创建新用户
     * 添加一个新的用户信息到系统
     *
     * @param userInfo 用户信息对象
     * @return 创建结果，包含新用户信息
     */
    @PostMapping("/add")
    public Mono<Result<UserInfo>> add(@RequestBody UserInfo userInfo) {
        return Mono.just(Result.success(userInfo));
    }

    /**
     * 更新用户信息
     * 更新指定用户的所有信息
     *
     * @param userInfo 更新后的用户信息
     * @return 更新结果
     */
    @PutMapping("/update")
    public Mono<IResult> update(@ModelAttribute UserInfo userInfo) {
        return Mono.just(Result.success(userInfo));
    }

    /**
     * 获取用户列表
     * 查询系统中所有用户的信息
     *
     * @return 用户列表
     */
    @RequestMapping(value = "/list", method = RequestMethod.GET)
    public Mono<IResult> list() {
        UserInfo userInfo = new UserInfo();
        userInfo.setId(1L);
        userInfo.setName("Tom");
        userInfo.setAge(25);
        return Mono.just(Result.success(Collections.singletonList(userInfo)));
    }

    /**
     * 删除用户
     * 根据用户ID删除指定用户的信息
     *
     * @param id 要删除的用户ID
     * @return 删除结果，包含被删除的用户信息
     */
    @DeleteMapping("/{id}")
    public Mono<IResult> delete(@PathVariable("id") Long id) {
        UserInfo userInfo = new UserInfo();
        userInfo.setId(id);
        userInfo.setName("Tony Stark");
        userInfo.setAge(45);
        return Mono.just(Result.success(userInfo));
    }

}

File: en/kotlin-spring-demo/src/main/kotlin/com/itangcent/springboot/kotlin/demo/controller/UserCtrl.kt
package com.itangcent.springboot.kotlin.demo.controller

import com.itangcent.common.dto.IResult
import com.itangcent.common.dto.Result
import com.itangcent.common.model.UserInfo
import org.springframework.web.bind.annotation.*

/**
 * User Management Controller
 * Provides basic CRUD operation interfaces for user information
 *
 * @module user
 */
@RestController
@RequestMapping(value = ["user"])
class UserCtrl {

    /**
     * Welcome Interface
     * Returns a simple welcome message to test if the service is running normally
     *
     * @return Returns "hello world" string
     */
    @RequestMapping(value = ["/index"])
    fun greeting(): String {
        return "hello world"
    }

    /**
     * Update Username Interface (Deprecated)
     * Updates username and personal signature by specified user ID
     * Recommended to use the new update method instead
     *
     * @param id User ID, used to uniquely identify the user
     * @param newName New username
     * @param slogon Personal signature, default value is "haha"
     * @param times Operation count, default value is 10
     * @return Returns updated user information
     * @deprecated Please use [UserCtrl.update] method instead
     */
    @RequestMapping(method = [RequestMethod.PUT], value = ["/set"])
    @Deprecated("Use [UserCtrl.update] instead")
    fun set(id: Long,
            @RequestParam newName: String,
            @RequestParam(required = false, defaultValue = "haha") slogon: String,
            @RequestParam(required = false, defaultValue = "10") times: Long): Any {
        val userInfo = UserInfo()
        userInfo.id = id
        userInfo.name = newName
        userInfo.age = 45
        return Result.success(userInfo)
    }

    /**
     * Get User Information Interface (Deprecated)
     * Gets detailed user information by user ID
     *
     * @param id User ID, used to uniquely identify the user
     * @return Returns response object containing detailed user information
     * @deprecated This interface is deprecated, please use the new interface instead
     */
    @Deprecated("")
    @GetMapping("/get/{id}")
    fun get(@PathVariable("id") id: Long?): IResult<UserInfo> {
        val userInfo = UserInfo()
        userInfo.id = id
        userInfo.name = "Tony Stark"
        userInfo.age = 45
        return Result.success(userInfo)
    }

    /**
     * Add New User Interface
     * Creates a new user information record
     *
     * @param userInfo User information object containing detailed user information
     * @return Returns the successfully created user information
     */
    @PostMapping("/add")
    fun add(@RequestBody userInfo: UserInfo): Result<UserInfo> {
        return Result.success(userInfo)
    }

    /**
     * Update User Information Interface
     * Updates information for an existing user
     *
     * @param userInfo User information object containing information to be updated
     * @return Returns the updated user information
     */
    @PutMapping("update")
    fun update(@ModelAttribute userInfo: UserInfo): Result<UserInfo> {
        return Result.success(userInfo)
    }

    /**
     * Get User List Interface
     * Returns user list filtered by user type
     *
     * @param type User type, refer to [com.itangcent.common.constant.UserType]
     *             1: Administrator
     *             2: Regular Member
     *             3: Guest
     * @return Returns list of users matching the criteria
     */
    @RequestMapping(value = ["/list"], method = [RequestMethod.GET])
    fun list(type: Int?): IResult<List<UserInfo>> {
        val userInfo = UserInfo()
        userInfo.id = 1L
        userInfo.name = "Tom"
        userInfo.age = 25
        return Result.success(listOf(userInfo))
    }

    /**
     * Get User List by Type Interface
     * Returns user list filtered by type specified in path parameter
     *
     * @param type User type, refer to [com.itangcent.common.constant.UserType]
     *             1: Administrator
     *             2: Regular Member
     *             3: Guest
     * @return Returns list of users of the specified type
     */
    @RequestMapping(value = ["/list/{type}"], method = [RequestMethod.GET])
    fun listTypeInPath(@PathVariable("type") type: Int?): IResult<List<UserInfo>> {
        val userInfo = UserInfo()
        userInfo.id = 1L
        userInfo.name = "Tom"
        userInfo.age = 25
        userInfo.type = type!!
        return Result.success(listOf(userInfo))
    }

    /**
     * Delete User Interface
     * Deletes user information by user ID
     *
     * @param id User ID to delete
     * @return Returns the deleted user information
     */
    @DeleteMapping("/{id}")
    fun delete(@PathVariable("id") id: Long?): Any {
        val userInfo = UserInfo()
        userInfo.id = id
        userInfo.name = "Tony Stark"
        userInfo.age = 45
        return Result.success(userInfo)
    }
}
